ui/src/ducks/app/monitoring.js

```diff
@@ -121,14 +128,32 @@ export function* fetchClusterStatus() {
     clusterHealth.apiServerStatus = getClusterQueryStatus(results[0]);
     clusterHealth.kubeSchedulerStatus = getClusterQueryStatus(results[1]);
     clusterHealth.kubeControllerManagerStatus = getClusterQueryStatus(
-      results[2]
+      results[2],
     );
     yield put(setPrometheusApiAvailable(true));
   } else {
+    const prometheusPod = yield call(
+      ApiK8s.getPodInNamespace,
+      'metalk8s-monitoring',
+      'prometheus',
+    );
+
+    if (!prometheusPod.error) {
+      const prometheusPodCondition =
+        prometheusPod?.body?.items[0]?.status?.conditions[0]?.type;
+      if (prometheusPodCondition === 'PodScheduled') {
```

Comment:
The status of the condition is `False` so I think it's better to check the type of condition?

Reply:
Mmh, actually this is not super robust. If one day another `condition` is added to _Pod_ `status`es, this will become somehow flaky. Instead, try something like:

```javascript
const conditions = prometheusPod?.body?.items[0]?.status?.conditions;
const scheduledCondition = conditions?.find(c => c.type === 'PodScheduled');
if (!scheduledCondition.status) {
  // update state with `scheduledCondition.message`
} else {
  // that's an unknown situation, handling TBD
}
```

Reply:
Indeed, you have a status for each type of the condition. you should filter `prometheusPod?.body?.items[0]?.status?.conditions` to get the "Ready" (or "PodScheduled" ?)condition instead of getting the first condition. Then check the status of this condition if it is `false` then store the message